src/tests/NXTTest.cpp

```diff
@@ -135,20 +140,50 @@ void NXTTest::SetUp() {
     nxtProcTable backendProcs;
     mBinding->GetProcAndDevice(&backendProcs, &backendDevice);
 
-    nxtSetProcs(&backendProcs);
-    device = nxt::Device::Acquire(backendDevice);
+    // Choose whether to use the backend procs and devices directly, or set up the wire.
+    nxtDevice cDevice = nullptr;
+    nxtProcTable procs;
+
+    if (gTestUsesWire) {
+        mC2sBuf = new nxt::wire::TerribleCommandBuffer();
+        mS2cBuf = new nxt::wire::TerribleCommandBuffer();
+
+        mWireServer = nxt::wire::NewServerCommandHandler(backendDevice, backendProcs, mS2cBuf);
+        mC2sBuf->SetHandler(mWireServer);
+
+        nxtDevice clientDevice;
+        nxtProcTable clientProcs;
+        mWireClient = nxt::wire::NewClientDevice(&clientProcs, &clientDevice, mC2sBuf);
+        mS2cBuf->SetHandler(mWireClient);
+
+        procs = clientProcs;
+        cDevice = clientDevice;
+    } else {
+        procs = backendProcs;
+        cDevice = backendDevice;
+    }
+
+    // Set up the device and queue because all tests need them, and NXTTest needs them too for the
+    // deferred expectations.
+    nxtSetProcs(&procs);
+    device = nxt::Device::Acquire(cDevice);
     queue = device.CreateQueueBuilder().GetResult();
 
+    // The swapchain isn't used by test but is useful when debugging with graphics debuggers that
```

Comment:
Nit: "used by tests" or "used by this test" instead of "used by test".

Reply:
Done